hazelcast/src/main/java/com/hazelcast/internal/partition/PartitionRuntimeState.java

```diff
@@ -35,7 +35,7 @@
 
 public final class PartitionRuntimeState implements IdentifiedDataSerializable {
 
-    // used for writing state
+    /** Map from address to the replica index of the first partition where the address occurrs. Used for writing state. */
     private Map<Address, Integer> addressToIndexes;
 
     // used for reading state
```

Comment:
Sorry, took me a while to find this comment again. That's a good question :)

Normally I use JavaDoc for everything that is non-private, regardless if it's public API or an internal class. Then it's usable via IDEA to help external and internal devs when using a class, method or constant while writing code.

For private code I tend to use single-line comments, since the JavaDoc won't be "rendered" in IDEA anyway. Also single-line JavaDoc will be re-formatted to three lines with the default code formatter of IDEA. So I'm not using them just to safe space, since eventually someone may use a code formatter on that class. An exception is private code which needs more than one sentence or `{@link}`.

But that's all just based on my gut feeling, I think there is no fixed rule or style guide for that.